design/one-pager-change-logs.md

```diff
@@ -0,0 +1,310 @@
+compared to observing the external system and recording the "after" state
+immediately after the operation as well. If we find during implementation and
+testing that recording the "after" state places too much burden on the system,
+we can revisit this decision.
+
+An additional information field is available in each change log entry for
+Providers to store any provider specific data they choose. Providers are not
+required to store any data at all in this field, as it is optional. As an
+example of data that could be stored here, `provider-aws` could store the
+specific endpoints it called to perform the given operation captured in the
+change log entry.
+
+### Generating Change Logs
+
+It is a goal that the change log data is generated as close to the source of
+truth as possible. All Crossplane providers, both [classic
+providers](https://github.com/crossplane/provider-template) and [Upjet generated
+providers](https://github.com/crossplane/upjet-provider-template) make use of
+the crossplane-runtime [managed
+reconciler](https://github.com/crossplane/crossplane-runtime/blob/master/pkg/reconciler/managed/reconciler.go),
+which calls into each provider to do specific CRUD operations on external
+resources.  Therefore, the managed reconciler is the ideal place to generate the
+change log entries very close to the source of truth.
+
+The managed reconciler calls the provider's [`ExternalClient.Observe()`
+method](https://github.com/crossplane/crossplane-runtime/blob/release-1.16/pkg/reconciler/managed/reconciler.go#L914)
+before any CRUD operation is performed, in order to understand the current state
+of the external system before the provider acts upon it. This up to date
+external state will be used to populate the "before" field in the change log
+entry.
+
+If a CRUD operation is performed later on in the same `Reconcile()`, then we
+will perform an additional call to this same `Observe()` method directly after
+the CRUD operation is performed. This second `Observe()` result will be used to
+populate the "after" field in the change log entry.
+
+Some helpful pointers to where CRUD operations are performed in the managed
+reconciler:
+
+* [Create](https://github.com/crossplane/crossplane-runtime/blob/release-1.16/pkg/reconciler/managed/reconciler.go#L1058)
+* [Update](https://github.com/crossplane/crossplane-runtime/blob/release-1.16/pkg/reconciler/managed/reconciler.go#L1189)
+* [Delete](https://github.com/crossplane/crossplane-runtime/blob/release-1.16/pkg/reconciler/managed/reconciler.go#L954)
+
+The code path through the managed reconciler's `Reconcile()` method are already
+a bit complex. One design choice that has helped minimize the complexity and
+assist in readability is that the reconciler returns as early as it can after
+performing an operation. Since we will now be adding an additional `Observe()`
+call after each CRUD operation, this will increase the complexity further, but
```

Comment:
don't cloud providers return the new state of the world after a CRUD operation? How many do?